api/envoy/extensions/http/original_ip_detection/custom_header/v3/custom_header.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package envoy.extensions.http.original_ip_detection.custom_header.v3;
+
+import "envoy/type/v3/http_status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.http.original_ip_detection.custom_header.v3";
+option java_outer_classname = "CustomHeaderProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Custom header original IP detection extension]
+
+// This extension allows for the original downstream remote IP to be detected
+// by reading the value from a configured header name. If the value is successfully parsed
+// as an IP, it'll be treated as the effective downstream remote address and seen as such
+// by all filters. See :ref:`original_ip_detection_extensions
+// <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.original_ip_detection_extensions>`
+// for an overview of how extensions operate and what happens when an extension fails
+// to detect the remote IP.
+//
+// [#extension: envoy.http.original_ip_detection.custom_header]
+message CustomHeaderConfig {
+  // Configuration options for rejecting requests when detection fails.
+  message RejectOptions {
+    // If reject_request_if_detection_fails is set, this sets the HTTP status that is sent downstream
+    // when detection fails. The default status is HTTP 403 Forbidden.
+    //
+    // .. note::
+    //   If this is set to < 400 or > 511, the default status 403 will be used instead.
+    type.v3.HttpStatus status_on_error = 1;
+
+    // If reject_request_if_detection_fails is set, this sets the HTTP body in the response
+    // when detection fails. If not set, an empty body is used.
+    string body_on_error = 2;
```

Comment:
I kind of feel this overlaps with the `LocalReplyConfig` in HCM somewhat:
1. Should we rely on that instead?
2. Should we at least have parity, e.g. model the body as a `DataSource`?

Reply:
Note that this is mostly an example extension and it's very unlikely someone will use as is (see the back and forth between Antonio's initial suggestion to make this an actual extension instead of dropping it along with tests and Alyssa's reviews around how this part of the config API -- for this filter -- should look).

Reply:
Why is it very unlikely anyone would use this? In general, if something is in the API, folks will consume :)

Reply:
Now that I think of this, we can just get rid of this field (that was Alyssa's initial suggestion -- should have listened :-).